src/store/reducers/deploymentsReducer.js

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+import 'rxjs';
+import { Observable } from 'rxjs';
+import moment from 'moment';
+import { schema, normalize } from 'normalizr';
+import update from 'immutability-helper';
+import { createSelector } from 'reselect';
+import { IoTHubManagerService } from 'services';
+import {
+  createReducerScenario,
+  createEpicScenario,
+  errorPendingInitialState,
+  pendingReducer,
+  errorReducer,
+  setPending,
+  resetPendingAndErrorReducer,
+  getPending,
+  getError,
+  toActionCreator
+} from 'store/utilities';
+
+// ========================= Epics - START
+const handleError = fromAction => error =>
+  Observable.of(redux.actions.registerError(fromAction.type, { error, fromAction }));
+
+export const epics = createEpicScenario({
+  /** Loads Deployments */
+  fetchDeployments: {
+    type: 'DEPLOYMENTS_FETCH',
+    epic: fromAction =>
+      IoTHubManagerService.getDeployments()
+        .map(toActionCreator(redux.actions.updateDeployments, fromAction))
+        .catch(handleError(fromAction))
+  },
+  /** Create a new deployment */
+  createDeployment: {
+    type: 'DEPLOYMENTS_CREATE',
+    epic: fromAction =>
+      IoTHubManagerService.createDeployment(fromAction.payload)
+        .map(toActionCreator(redux.actions.insertDeployment, fromAction))
+        .catch(handleError(fromAction))
+  },
+  /** Delete deployment */
+  deleteDeployment: {
+    type: 'DEPLOYMENTS_DELETE',
+    epic: fromAction =>
+      IoTHubManagerService.deleteDeployment(fromAction.payload)
+        .map(toActionCreator(redux.actions.deleteDeployment, fromAction))
+        .catch(handleError(fromAction))
+  }
+});
+// ========================= Epics - END
+
+// ========================= Schemas - START
+const deploymentSchema = new schema.Entity('deployments');
+const deploymentListSchema = new schema.Array(deploymentSchema);
+// ========================= Schemas - END
+
+// ========================= Reducers - START
+const initialState = { ...errorPendingInitialState, entities: {} };
+
+const insertDeploymentReducer = (state, { payload, fromAction }) => {
+  const { entities: { deployments }, result } = normalize(payload, deploymentSchema);
+  if (state.entities) {
+    return update(state, {
+      entities: { $merge: deployments },
+      items: { $splice: [[state.items.length, 0, result]] },
```

Comment:
Please insert at the beginning to be consistent with new-ish changes requested by UX.
```js
items: { $splice: [[0, 0, result]] },
```

Reply:
done